lib/util/http/auth.ts

```diff
@@ -36,7 +36,9 @@ export function applyAuthorization(inOptions: GotOptions): GotOptions {
         options.headers.authorization = `Bearer ${options.token}`;
       }
     } else {
-      options.headers.authorization = `Bearer ${options.token}`;
+      // Custom Auth type, eg `Basic XXXX_TOKEN`
+      const type = options.context?.authType ?? 'Bearer';
```

Comment:
```suggestion
      const type = options.context?.authType ?? 'Bearer';
```

I think we'll need to keep this here even if we add a `default` in config. IIRC defaults aren't passed through to nested configs like hostRules.

Reply:
I'll leave it